sql/decimal.go

```diff
@@ -128,20 +128,12 @@ func (t decimalType) Compare(a interface{}, b interface{}) (int, error) {
 	if err != nil {
 		return 0, err
 	}
-	ad, err := t.BoundsCheck(af.Decimal)
```

Comment:
we're sure we don't need bound checks?

Reply:
it uses one(seems to be the left side) of the two decimal values and gets its type are does bounds check on both the values using that type, which in some cases it can be smaller decimal precision and scale compared to the other value's precision and scale, so it seems redundant to check bounds for these values when comparing two values.